library/Vanilla/Addon.php

```diff
@@ -448,26 +449,68 @@ private function scanClasses() {
         return $classes;
     }
 
+    /**
+     * Scan the addon for potential class paths.
+     *
+     * @return \Traversable Returns a list of paths to PHP files.
+     */
     private function scanClassPaths() {
-        $globs = [
-            '/*.php',
-            '/controllers/*.php',
-            '/library/*.php',
-            '/models/*.php',
-            '/modules/*.php',
+        $dirs = [
+            '',
+            '/controllers',
+            '/Controllers',
+            '/library',
+            '/src',
+            '/models',
+            '/Models',
+            '/modules',
+            '/Modules',
             '/settings/class.hooks.php'
         ];
 
         $result = [];
-        foreach ($globs as $glob) {
-            $paths = $this->glob($glob);
-            $result = array_merge($result, $paths);
+        foreach ($dirs as $dir) {
+            foreach ($this->scanDirPhp($dir) as $path) {
+                yield $path;
+            }
         }
 
         return $result;
```

Comment:
Non-empty return statements do not appear to be valid in PHP, prior to v7. This is also true with HHVM. The build is currently failing with the following error:

> Generators cannot return values using "return"

From PHP's [Generator Syntax docs page](http://php.net/manual/en/language.generators.syntax.php):

> A generator cannot return a value: doing so will result in a compile error. An empty return statement is valid syntax within a generator and it will terminate the generator.